vm/src/org/mozartoz/truffle/nodes/builtins/NumberBuiltins.java

```diff
@@ -61,6 +64,52 @@ long neg(long value) {
 
 	}
 
+	@Builtin(name = "pow", deref = ALL)
+	@GenerateNodeFactory
+	@NodeChildren({ @NodeChild("left"), @NodeChild("right") })
+	public static abstract class PowNode extends OzNode {
+
+		@Specialization(guards = { "isInteger(base)", "exponent <= 94", "exponent == cachedExponent" })
```

Comment:
Done for the `cachedExponent`

To me, a faster specialization is useful as long as the operation does not overflow. So basically, 63 would have been my bound. However, it produces 11 nodes when unrolled. So I just took the highest exponent such that no more than 11 nodes could be unrolled. 94 happens to be that number.

> 59 10
> 60 9
> 61 10
> 62 10
> 63 11
> 64 7
> 65 8
> 66 8
> 67 9
> 68 8
> 69 9
> 70 9
> 71 10
> 72 8
> 73 9
> 74 9
> 75 10
> 76 9
> 77 10
> 78 10
> 79 11
> 80 8
> 81 9
> 82 9
> 83 10
> 84 9
> 85 10
> 86 10
> 87 11
> 88 9
> 89 10
> 90 10
> 91 11
> 92 10
> 93 11
> 94 11
> 95 12